Security/HttpAuthentication.php

```diff
@@ -52,6 +52,6 @@ public function transformRequest(Request $request) {
         array_push($headers, 'Authorization: Basic ' . base64_encode($this->config['user'] . ':' . $this->config['password']));
         $options[CURLOPT_HTTPHEADER] = $headers;
 
-        return new Request($request->getMethod(), $request->getUrl(), $options, $request->getQuery(), $request->getPayload());
+        return new Request($request->getMethod(), $request->getUrl(), $options, $request->getQuery(), $request->getPayload(), $request->getExpectedStatusCode());
```

Comment:
die api hier is quatsch in meinen augen. soweit ich das sehe, setzt du doch nur die options auf dem ding, ansonsten bleibt der request gleich, also warum nich `return $request->setOptions($options)` und dann den konstruktor so überladen, dass der nich eine millionen argumente brauch :D, also intern iwie sowas:

```js
constructor(array raw) {
     this.method = raw.method;
     this.options  = raw.options
     ...
}

setOptions(options) {
    return new Request([
        method: this.method,
        options: options,
        url:         this.url
        ...
    ]);
}
```